### PR TITLE
Adds certificate authentication as an optional credential parameter

### LINKED ISSUE #23: Authentication with certificate
I'm transferring this issue over, as I previously incorrectly posted to the presto adapter repo. 

As far as I can tell, there is no way to pass in a path to a certificate for an SSL connection to the API. I believe this needs to be set on the trino connection that is created.

https://github.com/starburstdata/dbt-trino/blob/afa4091f971d264aad3bfe51476fd1497a4ea60e/dbt/adapters/trino/connections.py#L185

### PR DESCRIPTION
**Pull request by Haunfelder**
Resolves #23

### CODE REVIEW

**Comment by hovaesco** [comment]
Please add a test for it.

**Comment by Haunfelder** [comment]
I'm uncertain whether the added test is sufficient, so let me know if there's anything you would like me to add.

File: README.md [hunk 1]
```diff
@@ -46,6 +46,7 @@ A dbt profile can be configured to run against Trino using the following configu
 | password           | Password for authentication                                                                                  | Optional (required if `method` is `ldap` or `kerberos`)                            | `none` or `abc123`               |
 | http_headers       | HTTP Headers to send alongside requests to Trino, specified as a yaml dictionary of (header, value) pairs.   | Optional                                                                           | `X-Trino-Client-Info: dbt-trino` |
 | http_scheme        | The HTTP scheme to use for requests to Trino                                                                 | Optional (default is `http`, or `https` for `method: kerberos` and `method: ldap`) | `https` or `http`                |
+| cert               | The full path to a certificate file for authentication with trino          | Optional  |                 |
```
Comment (findinpath) [inline README.md:49]: nit: could you please align the columns in this line accordingly to the other rows?

File: dbt/adapters/trino/connections.py [hunk 2]
```diff
@@ -196,7 +197,8 @@ def open(cls, connection):
             isolation_level=IsolationLevel.AUTOCOMMIT,
             source="dbt-trino",
         )
-        connection.state = "open"
+        trino_conn._http_session.verify = credentials.cert
```
Comment (findinpath) [inline dbt/adapters/trino/connections.py:200]: I had a look at https://docs.python-requests.org/en/latest/user/advanced/#client-side-certificates and the certificate seems to be set differently. I'm not quite sure that you're setting the right field here.

**Review by findinpath — CHANGES_REQUESTED** [review]
Please review the usage of `requests` library for setting the certificate path.

File: dbt/adapters/trino/connections.py (hunk 2 above)
Comment (hovaesco) [reply dbt/adapters/trino/connections.py:200]: It looks correct based on: 
https://github.com/trinodb/trino-python-client/pull/31

File: README.md (hunk 1 above)
Comment (findinpath) [inline README.md:49]: please change the naming to something more meaningful `client_certificate_path` / `certificate_path`

Comment (hovaesco) [reply README.md:49]: It should be changed it in trino-python-client first and then change it in dbt-trino. Now all parameters between these two are aligned.

**Comment by hovaesco** [comment]
@Haunfelder thanks for the contribution!

**Pull request merged**